src/gstreamer/gstkvssink.cpp

```diff
@@ -1148,9 +1181,14 @@ gst_kvs_sink_handle_buffer (GstCollectPads * pads,
         buf->pts += data->producer_start_time - data->first_pts;
     }
 
-    put_frame(data->kinesis_video_stream, info.data, info.size,
+    putFrameSuccess = put_frame(kvssink, info.data, info.size,
               std::chrono::nanoseconds(buf->pts),
               std::chrono::nanoseconds(buf->dts), kinesis_video_flags, track_id, data->frame_count);
+
+    if(kvssink->data->onFirstFrame && putFrameSuccess){
+        kvssink->data->onFirstFrame = false;
```

Comment:
You can update this in put_frame function itself and remove this block. Right now, we dont do anything if putFrame failed, would say we should log if putFrameSuccess is false to track putFrame failure. So the log would contain information on frame index, kinesis_video_flags etc.